### PR TITLE
handle proto.DataplaneInSync msg in status_reporter

### PR DESCRIPTION
**Pull request by aaaaaaaalex**
## Description

A panic is occurring on OpenStack Mainline ST due to an unhandled protobuf type recently added 

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/statusrep/status_reporter.go [hunk 1]
```diff
@@ -182,6 +182,8 @@ loop:
 					Hostname:   esr.hostname,
 					EndpointID: msg.Id.EndpointId,
 				}
+			case *proto.DataplaneInSync:
```
Comment (fasaxc) [inline felix/statusrep/status_reporter.go:185]: Could we remove the old in-sync channel completely now?  This should be better.

File: felix/fv/status_report_test.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+			return statuses.KVPairs
+		}
+
+		It("should write endpoint to etcd", func() {
+			Eventually(getStatuses, "10s").Should(HaveLen(1))
+			kvs := getStatuses()
+			wepStatusKey, ok := kvs[0].Key.(model.WorkloadEndpointStatusKey)
+			Expect(ok).To(BeTrue(), "Unexpected key type when listing workload endpoint statuses")
+			Expect(wepStatusKey.Hostname).To(BeEquivalentTo(tc.Felixes[0].Hostname))
```
Comment (fasaxc) [inline felix/fv/status_report_test.go:96]: What does BeEquivalentTo() do vs Equal() (I tend to use the latter)

**Review by fasaxc — COMMENTED** [review]
Looking good, I think I'd remove the old in-sync channel too, WDYT?

File: felix/statusrep/status_reporter.go (hunk 1 above)
Comment (aaaaaaaalex) [reply felix/statusrep/status_reporter.go:185]: I guess so, but avoided out of nerves of breaking something else :laughing: WIll give it a shot

File: felix/fv/status_report_test.go (hunk 2 above)
Comment (aaaaaaaalex) [reply felix/fv/status_report_test.go:96]: BeEquivalentTo neglects to check types are the same, as I understand it. Still checks values align, not sure of the specifics beyond that. Equal() would work here too

File: felix/statusrep/status_reporter.go (hunk 1 above)
Comment (aaaaaaaalex) [reply felix/statusrep/status_reporter.go:179]: Done - will monitor CI in-case I broke anything else :laughing:

**Review by fasaxc — APPROVED** [review]
LGTM

**Pull request merged**